src/TransactionController.ts

```diff
@@ -354,30 +352,31 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
 	}
 
 	/**
-	 * Resiliently checks all submitted transactions for confirmations in all
-	 * blocks maintained in a sibling BlockHistoryController
+	 * Resiliently checks all submitted transactions on the blockchain
+	 * and verifies that it has been included in a block
+	 * when that happens, the tx status is updated to confirmed
 	 *
 	 * @returns - Promise resolving when this operation completes
 	 */
 	async queryTransactionStatuses() {
 		const { transactions } = this.state;
-		const blockHistory = this.context.BlockHistoryController as BlockHistoryController;
-		const network = this.context.NetworkController as NetworkController;
+		const network = this.context.NetworkController;
 		const currentNetworkID = network.state.network;
-
-		const confirmedHashes = blockHistory.state.recentBlocks
-			.reduce((hashes: string[], block) => {
-				return hashes.concat(block.transactions);
-			}, [])
-			.map((transaction: any) => transaction && transaction.hash);
-
-		transactions.forEach((meta, index) => {
-			const isConfirmed = confirmedHashes.indexOf(meta.transactionHash!) > -1;
-			if (meta.networkID === currentNetworkID && meta.status === 'submitted' && isConfirmed) {
-				transactions[index].status = 'confirmed';
-				this.hub.emit(`${meta.id}:confirmed`, meta);
+		transactions.forEach(async (meta, index) => {
```

Comment:
I think we need to use a `for...of` loop here, otherwise `this.update({ transactions: [...transactions] });` will get called before each transaction status query completes. We could also parallelize all queries using `Promise.all` to speed things up. Either way, we need to wait for everything to be done.

Reply:
Makes sense. I went with Promise.all as you suggested.